lms/djangoapps/instructor_task/models.py

```diff
@@ -101,6 +102,16 @@ def create(cls, course_id, task_type, task_key, task_input, requester):
         task_id = str(uuid4())
         json_task_input = json.dumps(task_input)
 
+        # check length of task_input, and return an exception if it's too long
+        if len(json_task_input) > TASK_INPUT_LENGTH:
```

Comment:
any specific reason this is not implemented inside `save_now()` method?

Reply:
just for an early exit if there is length validation failure

Reply:
In my case, you have to dup the logic everywhere you are creating an instance, let's have the more centralized logic inside `save_now`, if that doesn't cause other complications.

Reply:
We are using `create` method to create an instance of `InstructorTask`. I believe this method is purposefully written to do creation work leaving `save` method to be called in `@transaction.atomic` context. Please let me know if you think we should move this and other implementation of creation in `save` method.

Reply:
What do you think about the workaround for using context manager `@transaction.atomic`?

Reply:
Sorry, I didn't get why we would need a workaround for `@transaction.atomic`?

Reply:
Sorry, I wasn't so clear.

I was asking for your opinion about moving the validation inside the `save_now` method and using the context manager `@transaction.atomic` above `self.save()`?

Reply:
Please have a look at the signature of `create` method. If we don't want a `create` method we will have to change the signature of `__init__` method, which I believe is not a good practice. 
As I said earlier, it is just an early exit if we have a validation issue in the passed argument.
Please see note at https://docs.djangoproject.com/en/2.2/ref/models/instances/